github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs

```diff
@@ -0,0 +1,53 @@
+use serde_json::json;
+use flows_connector_dsi::{github::inbound};
+use wasmedge_bindgen_macro::*;
+
+#[wasmedge_bindgen]
+pub fn run(s: String) -> Result<String, String> {
+    #[allow(unused_imports)]
+    use wasmedge_bindgen;
+    _run(s)
+}
+
+fn _run(s: String) -> Result<String, String> {
+    let payload = inbound(s)?;
+    
+    match payload.pull_request {
+        None => { return Err("no pull request found".to_string()); },
+        Some(_) => {},
+    }
```

Comment:
In fact `pull_request` field is unnecessary in your program.
btw, if you want to check whether `pull_request` is exists, you should use:

```rust
payload.get_pull_request()?;
```

`get_xxx()` will return a `Result` which including the error message, and than `?` operator will return when it was 'Err()'

Reply:
nice